src/Sarif.Multitool/Rules/RuleResources.resx

```diff
@@ -210,6 +210,18 @@
   <data name="SARIF1019_RuleIdMustBePresentAndConsistent" xml:space="preserve">
     <value>In every result, at least one of the properties result.ruleId and result.rule.id must be present. If both are present, they must be equal.</value>
   </data>
+  <data name="SARIF1022_PreferMessageIdInsteadOfMessageText" xml:space="preserve">
+    <value>Prefer message.id instead of message.text</value>
+  </data>
+  <data name="SARIF1022_ResultShouldUseMessageIdInsteadOfMessageText" xml:space="preserve">
+    <value>{0}: The result should use message.id instead of message.text value '{1}'</value>
+  </data>
+  <data name="SARIF1022_RuleMetadataDoesntIncludeMessageId" xml:space="preserve">
+    <value>{0}: The rule metadata doesn't include the message id value '{1}'</value>
+  </data>
+  <data name="SARIF1022_RuleMetadataIsNotConsistentWithArguments" xml:space="preserve">
+    <value>{0}: The rule metadata text '{1}' contains invalid quantity of arguments '{2}'</value>
```

Comment:
>{0}: The rule metadata text '{1}' contains invalid quantity of arguments '{2}' [](start = 11, length = 78)

"{0}: The result message contains {1} message arguments for the message string with id '{2}', but the metadata for rule '{3}' defines message string '{2}' to take {4} arguments."

Reply:
the new message: {0}: The result message contains {1} message arguments for the message string with id '{2}', but the metadata for rule '{3}' defines message string '{2}' to take {4} arguments: '{5}'

0 = pointer
1 = result.message.arguments.count
2 = result.message.id
3 = result.ruleid ?? result.rule.id
4 = count of placeholders in text
5 = text itself

Reply:
For 4, it's not the count of placeholders. In the message you just presented, there are 7 placeholders, but only 6 of them are distinct. Placeholder {2} occurs twice in the message.

---
In reply to: [434725402](https://github.com/microsoft/sarif-sdk/pull/1901#discussion_r434725402) [](ancestors = 434725402)